R/jexcel.R

```diff
@@ -308,75 +296,37 @@ excelTable <-
       paramList$minDimensions <- minDimensions
     }
 
-    # Check columnSorting
-    if (!columnSorting) {
-      paramList$columnSorting = columnSorting
-
-    }
-
-    #Check columnDrag
-    if (columnDrag) {
-      paramList$columnDrag <- TRUE
-    }
-
-    # Check columnResize
-    if (!columnResize) {
-      paramList$columnResize <- FALSE
-    }
-
-    # Check rowResize
-    if (rowResize) {
-      paramList$rowResize <- TRUE
-    }
-
-    # Check rowDrag
-    if (!rowDrag) {
-      paramList$rowDrag <- FALSE
-    }
-
-    # Check editable
-    if (!editable) {
-      paramList$editable <- FALSE
-    }
-
-    # Check allowInsertRow
-    if (!allowInsertRow) {
-      paramList$allowInsertRow <- FALSE
-    }
-
-    # Check allowInsertColumn
-    if (!allowInsertColumn) {
-      paramList$allowInsertColumn <- FALSE
-    }
-
-    # Check allowDeleteRow
-    if (!allowDeleteRow) {
-      paramList$allowDeleteRow <- FALSE
-    }
-
-    # Check allowDeleteColumn
-    if (!allowDeleteColumn) {
-      paramList$allowDeleteColumn <- FALSE
-    }
-
-    # Check allowRenameColumn
-    if (!allowRenameColumn) {
-      paramList$allowRenameColumn <- FALSE
-    }
-
-    # Check allowComments
-    if (allowComments) {
-      paramList$allowComments <- TRUE
-    }
-
-    # Check wordWrap
-    if (wordWrap) {
-      paramList$wordWrap <- TRUE
-    }
-
-    # Check selectionCopy
-    if (!selectionCopy) {
-      paramList$selectionCopy <- FALSE
+    # Check logical arguments
+    for (arg in c(
+      "columnSorting",
+      "columnDrag",
+      "columnResize",
+      "rowResize",
+      "rowDrag",
+      "editable",
+      "allowInsertRow",
+      "allowInsertColumn",
+      "allowDeleteRow",
+      "allowDeleteColumn",
+      "allowRenameColumn",
+      "allowComments",
+      "wordWrap",
+      "selectionCopy",
+      "search",
+      "fullscreen",
+      "lazyLoading",
+      "loadingSpin"
+    )) {
+      argvalue <- get(arg)
+      if(!is.null(argvalue)) {
+        # now check these arguments to make sure they are logical
+        if(is.logical(argvalue)) {
+          paramList[[arg]] <- argvalue
+        } else {
+          warning("Argument ", arg, " should be either TRUE or FALSE.  Ignoring ", arg, ".", call. = FALSE)
+          paramList[[arg]] <- NULL
```

Comment:
Passing `NULL` means the argument is not passed to the widget and will use `jExcel` default.  Here is an example where each of these arguments is `NULL`.

```
library(excelR)

nullparams <- list(data = matrix(1:100,ncol=10))
for (arg in c(
  "columnSorting",
  "columnDrag",
  "columnResize",
  "rowResize",
  "rowDrag",
  "editable",
  "allowInsertRow",
  "allowInsertColumn",
  "allowDeleteRow",
  "allowDeleteColumn",
  "allowRenameColumn",
  "allowComments",
  "wordWrap",
  "selectionCopy",
  "search",
  "lazyLoading",
  "loadingSpin"
)) {
  nullparams[[arg]] = NULL
}

do.call(excelTable, nullparams)
```

![image](https://user-images.githubusercontent.com/837910/61535031-cbef0900-a9f6-11e9-9d03-0e1716f95e6b.png)
<img width="352" alt="2019-07-19" src="https://user-images.githubusercontent.com/837910/61535111-09ec2d00-a9f7-11e9-9e66-0139d321dfaf.png">

`jExcel` supplies all defaults in [lines](https://github.com/paulhodel/jexcel/blob/master/dist/jexcel.js#L24-L163) so empty is fine.

Reply:
I guess I am missing something here. Building your branch and running the above code you provided gives me the following error:
```
Error in !columnSorting : invalid argument type
In addition: Warning message:
In (function (data = NULL, columns = NULL, colHeaders = NULL, rowHeight = NULL,  :
  Since both column title and colHeaders are not specified 'data' column name will be used as column headers
```

Reply:
That is really strange, since `!columnSorting` occurs nowhere in my code.

Reply:
Yes I was wondering the same thing. The package I downloaded from CRAN also give me the same error. So I guess it has nothing to do with your code.

Reply:
Your approach seems to work.